src/sentry/api/endpoints/debug_files.py

```diff
@@ -49,6 +53,30 @@ def upload_from_request(request, project):
     return Response(serialize(files, request.user), status=201)
 
 
+def has_download_permission(request, project):
```

Comment:
_nit_: Since this is the exact same logic as for the `EventAttachmentsDetailsPermission`, it would be nice to factor out the common bits into a helper function that can perform these checks based on a required permission passed in as a parmeter.